src/lib.rs

```diff
@@ -142,80 +135,112 @@ struct CoreSDK<WP> {
     shutdown_requested: AtomicBool,
 }
 
+/// Can be used inside the CoreSDK impl to block on any method that polls the server until it
+/// responds, or until the shutdown flag is set (aborting the poll)
+macro_rules! abort_on_shutdown {
+    ($self:ident, $gateway_fn:tt, $poll_arg:expr) => {
+        $self.runtime.block_on(async {
+            let shutdownfut = async {
+                loop {
```

Comment:
Not with a channel, since shutdown can and will be read multiple times. I could use a condition variable here but it just didn't seem terribly important to optimize at the moment. I can fix this in another PR